### PR TITLE
Switch on String instead of cascading if-else

### PR DESCRIPTION
**Pull request by sanjanarampurkottur01**

### CODE REVIEW

**Comment by garydgregory** [comment]
@sanjanarampurkottur01 
Please don't change the formatting style, it makes PRs noisy and takes longer and is harder to review.

**Comment by sanjanarampurkottur01** [comment]
Hi Gary,

I am extremely sorry about that, I will change the formatting style and let
you know as soon as possible, once I push the changes.

Thank you.

Sincerely,
Sanjana Rampur Kottur

**Comment by sanjanarampurkottur01** [comment]
> @sanjanarampurkottur01 Please don't change the formatting style, it makes PRs noisy and takes longer and is harder to review.

@garydgregory , I will fix that andl et you know. I am extremely sorry for the inconvenience caused.

**Comment by sanjanarampurkottur01** [comment]
Hi Gary,

I have made the necessary changes to
https://github.com/apache/commons-dbutils/pull/220, I kindly request you to
review and approve the same.
My sincere apologies for the same.

Thank you.

Sincerely,
Sanjana Rampur Kottur

File: src/test/java/org/apache/commons/dbutils/MockResultSet.java [hunk 1]
```diff
@@ -21,10 +21,7 @@
 import java.sql.ResultSet;
 import java.sql.ResultSetMetaData;
 import java.sql.SQLException;
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.Iterator;
-import java.util.List;
+import java.util.*;
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/dbutils/MockResultSet.java:24]: Don't change to start imports.

Comment (sanjanarampurkottur01) [reply src/test/java/org/apache/commons/dbutils/MockResultSet.java:24]: Sure @garydgregory , 

I'm currently addressing the issue, and I appreciate your prompt review. 

Thank you.

Comment (garydgregory) [reply src/test/java/org/apache/commons/dbutils/MockResultSet.java:24]: and please rebase on master ;-) TY!

File: src/test/java/org/apache/commons/dbutils/MockResultSet.java [hunk 2]
```diff
@@ -265,64 +266,86 @@ protected String getString(final int columnIndex) throws SQLException {
...
+        } else if (isNonColumnMethod(methodName)) {
+            return handleNonColumnMethod(methodName, proxy, args);
+        }
+        throw new UnsupportedOperationException("Unsupported method: " + methodName);
+    }
 
-        } else if (methodName.equals("getString")) {
-            return this.getString(columnIndex(args));
+    // Define methods for handling specific cases
+    private Object handleGetMetaData() throws SQLException {
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/dbutils/MockResultSet.java:292]: This is superfluous IMO; just leave the implementation inlined as it was.

File: src/test/java/org/apache/commons/dbutils/MockResultSet.java [hunk 3]
```diff
@@ -265,64 +266,86 @@ protected String getString(final int columnIndex) throws SQLException {
...
+        return this.getMetaData();
+    }
 
-        } else if (methodName.equals("wasNull")) {
-            return this.wasNull();
+    private Object handleNext() throws SQLException {
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/dbutils/MockResultSet.java:296]: This is superfluous IMO; just leave the implementation inlined as it was.

**Review by garydgregory — CHANGES_REQUESTED** [review]
Hi @sanjanarampurkottur01 
Thank you for your update.
Please see my comments.

File: src/test/java/org/apache/commons/dbutils/MockResultSet.java (hunk 2 above)
Comment (sanjanarampurkottur01) [reply src/test/java/org/apache/commons/dbutils/MockResultSet.java]: Resolved it. Thank you :)

File: src/test/java/org/apache/commons/dbutils/MockResultSet.java (hunk 3 above)
Comment (sanjanarampurkottur01) [reply src/test/java/org/apache/commons/dbutils/MockResultSet.java]: Resolved this as well!!

File: src/test/java/org/apache/commons/dbutils/MockResultSet.java [hunk 4]
```diff
@@ -265,64 +266,77 @@ protected String getString(final int columnIndex) throws SQLException {
...
+                return this.wasNull();
+            default:
+                throw new UnsupportedOperationException("Unsupported column method: " + methodName);
+        }
+    }
 
-        } else if (methodName.equals("toString")) {
-            return "MockResultSet " + System.identityHashCode(proxy);
+    private boolean isNonColumnMethod(String methodName) {
+        Set<String> methodNames = Set.of("isLast", "hashCode", "toString", "equals");
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/dbutils/MockResultSet.java:323]: This Set does not need to be created on each invocation, just make it a constant and immutable.

**Review by garydgregory — CHANGES_REQUESTED** [review]
@sanjanarampurkottur01 
Thank you for your update.
Please see my comment.

File: src/test/java/org/apache/commons/dbutils/MockResultSet.java (hunk 4 above)
Comment (sanjanarampurkottur01) [reply src/test/java/org/apache/commons/dbutils/MockResultSet.java]: Modified this as well. Thanks for noticing it. Static field will be a better choice. Thank you!

File: src/test/java/org/apache/commons/dbutils/MockResultSet.java [hunk 5]
```diff
@@ -53,6 +54,8 @@ public static ResultSet create(final ResultSetMetaData metaData, final Object[][
 
     private Boolean wasNull = Boolean.FALSE;
 
+    private static final Set<String> METHOD_NAMES = Set.of("isLast", "hashCode", "toString", "equals");
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/dbutils/MockResultSet.java:57]: Make the Set immutable.

**Review by garydgregory — CHANGES_REQUESTED** [review]
-

**Pull request merged**